src/Task.ts

```diff
@@ -319,11 +315,30 @@ export function loadOrCreateAcceptanceList(tokensAcceptanceListId: string): Acce
   return acceptanceList
 }
 
+export function loadOrCreateTimelock(timelockId: string): Timelock {
+  let timelock = Timelock.load(timelockId)
+
+  if (timelock === null) {
+    timelock = new Timelock(timelockId)
+    timelock.allowedAt = BigInt.zero()
+    timelock.frequency = BigInt.zero()
+    timelock.window = BigInt.zero()
+  }
+
+  return timelock
+}
+
 export function parseAcceptanceType(op: i32): string {
   if (op == 0) return 'DenyList'
   else return 'AllowList'
 }
 
+export function parseTimelockMode(mode: i32): string {
+  if (mode == 0) return 'Seconds'
+  if (mode == 1) return 'On day'
+  else return 'Last month day'
```

Comment:
A good practice here is not to return the last value possible in any other case, we should do sth like:

```suggestion
   if (mode == 2) return 'LastMonthDay'
   else return 'Unknown'
```

and ofc we should add `Unknown` as a possible enum value in the definition. 

Note this way, if we support a new mode in the contracts, the frontend will alarm us about it, otherwise we may never detect it as it will treat it as "last month day"